File: 14/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int N = strs.size();
        if(N==1) return strs[0];
        int len = strs[0].length();
        for(int i=1; i<N; i++){
            if(len<strs[i].length()) len=strs[i].length();
        }

        int ans=0;
        for(int k=0; k<len; k++){
            int bad=0;
            for(int i=0; i<N; i++){
                if(strs[i][k]!=strs[0][k]){
                    bad=1;
                    break;
                }
            }
            if(bad==1)break;
            ans++;
        }
        return strs[0].substr(0, ans);
    }
};

File: 14/longest-common-prefix.py
# LeetCode 14. Longest Common Prefix
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        M = len(strs)
        N = min(len(strs[i]) for i in range(M))  # min(每條的長度)
        ans = 0  # 答案的長度

        for j in range(N):
            for i in range(1,M):
                if strs[i][j]!= strs[0][j]: # 有不一致，直接結束
                    return strs[0][:ans]
            ans += 1
        return strs[0][:ans]
